model/src/main/java/hu/blackbelt/judo/meta/asm/runtime/AsmUtilsCache.java

```diff
@@ -20,18 +20,35 @@
  * #L%
  */
 
-import org.eclipse.emf.ecore.EAttribute;
-import org.eclipse.emf.ecore.EClassifier;
-import org.eclipse.emf.ecore.EOperation;
-import org.eclipse.emf.ecore.EReference;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.eclipse.emf.ecore.*;
+import org.eclipse.emf.ecore.resource.ResourceSet;
 
 import java.util.Collection;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
 
 public class AsmUtilsCache {
 
+    private final static CacheLoader<ResourceSet, AsmUtilsCache> cacheLoader = new CacheLoader<>() {
+        @Override
+        public AsmUtilsCache load(ResourceSet resourceSet) {
+            AsmUtilsCache cache = new AsmUtilsCache();
+            return cache;
+        }
+    };
+
+    private final static LoadingCache<ResourceSet, AsmUtilsCache> cacheProvider = CacheBuilder
+            .newBuilder()
+            .expireAfterAccess(60, TimeUnit.SECONDS)
```

Comment:
Fixed value?

Reply:
Okay. I create a system parameter for that :)